model.py

```diff
@@ -7012,6 +7031,28 @@ def default_filename(self, link=None, destination_type=None):
             filename += extension
         return filename
 
+    @property
+    def external_media_type(self):
+        if self.clean_media_type == self.SVG_MEDIA_TYPE:
+            return self.PNG_MEDIA_TYPE
+        return self.media_type
+
+    def external_content(self):
+        """Return a filehandle to the representation's contents, as they
+        should be mirrored externally, and the media type to be used
+        when mirroring.
+        """
+        if not self.is_image or self.clean_media_type != self.SVG_MEDIA_TYPE:
+            # Passthrough
+            return self.content_fh()
+
+        # This representation is an SVG image. We want to mirror it as
+        # PNG.
+        image = self.as_image()
+        output = StringIO()
+        image.save(output, format='PNG')
+        return StringIO(output.getvalue())
```

Comment:
The S3 code calls read() on the filehandle and if you write to a StringIO then .read() returns the empty string. But calling .seek(0) first does the same thing more efficiently so I'll change it.